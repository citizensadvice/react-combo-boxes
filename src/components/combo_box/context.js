import { createContext } from 'react';
import { visuallyHiddenClassName } from '../../constants/visually_hidden_class_name';

export const Context = createContext({
  visuallyHiddenClassName,
  dispatch: null,
});
