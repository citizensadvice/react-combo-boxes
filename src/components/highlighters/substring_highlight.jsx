import React from 'react';
import { HighlightValue } from '../highlight_value';
import { substringHighlighter } from '../../highlighters/substring_highlighter';

export function SubstringHighlight(props) {
  return (
    <HighlightValue
      highlight={substringHighlighter}
      {...props}
    />
  );
}
