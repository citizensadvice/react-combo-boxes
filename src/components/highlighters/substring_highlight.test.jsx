import React from 'react';
import { render } from '@testing-library/react';
import { SubstringHighlight } from './substring_highlight';
import { Context } from '../../context';

function TestHighlight({ children, value, inverse, ...props }) {
  return (
    <Context.Provider value={{ ...props, props: { value } }}>
      <SubstringHighlight inverse={inverse}>
        {children}
      </SubstringHighlight>
    </Context.Provider>
  );
}

it('does not highlight with no children', () => {
  const { container } = render((
    <TestHighlight search="bar" />
  ));

  expect(container).toContainHTML('<div></div>');
});

it('does not highlight with no search children', () => {
  const { container } = render((
    <TestHighlight search={null}>
      foo
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo</div>');
});

it('does not highlight no match', () => {
  const { container } = render((
    <TestHighlight search="bar">
      foo
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo</div>');
});

it('highlights a substring', () => {
  const { container } = render((
    <TestHighlight search="bar">
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo <mark>bar</mark> foo bar</div>');
});

it('highlights an existing value substring', () => {
  const { container } = render((
    <TestHighlight value={{ label: 'bar' }}>
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div>foo <mark>bar</mark> foo bar</div>');
});

it('inverses the highlight', () => {
  const { container } = render((
    <TestHighlight search="bar" inverse>
      foo bar foo bar
    </TestHighlight>
  ));

  expect(container).toContainHTML('<div><mark>foo </mark>bar<mark> foo bar</mark></div>');
});
