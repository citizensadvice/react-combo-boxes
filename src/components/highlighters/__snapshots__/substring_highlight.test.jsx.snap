// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlights a substring 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    foo 
    <mark>
      bar
    </mark>
     foo bar
  </span>
</div>
`;

exports[`highlights an existing value substring 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    foo 
    <mark>
      bar
    </mark>
     foo bar
  </span>
</div>
`;

exports[`inverses the highlight 1`] = `
<div>
  <span
    class="sr-only"
  >
    foo bar foo bar
  </span>
  <span
    aria-hidden="true"
  >
    <mark>
      foo
    </mark>
     bar 
    <mark>
      foo bar
    </mark>
  </span>
</div>
`;
